usaspending_api/download/filestreaming/download_generation.py

```diff
@@ -111,14 +159,40 @@ def generate_download(download_job: DownloadJob, origination: Optional[str] = No
             append_files_to_zip_file([file_description_path], zip_file_path)
         download_job.file_size = os.stat(zip_file_path).st_size
     except InvalidParameterException as e:
-        exc_msg = "InvalidParameterException was raised while attempting to process the DownloadJob"
-        fail_download(download_job, e, exc_msg)
-        raise InvalidParameterException(e)
+        with SubprocessTrace(
+            name=f"job.{JOB_TYPE}.generate_download",
+            kind=SpanKind.INTERNAL,
+            service="bulk-download",
+        ) as error_span:
+            error_span.set_attributes(
+                {
+                    "service": "bulk-download",
+                    "span_type": "Internal",
+                    "message": "InvalidParameterException was raised while attempting to process the DownloadJob",
+                    "error": str(e),
+                }
+            )
+            exc_msg = "InvalidParameterException was raised while attempting to process the DownloadJob"
+            fail_download(download_job, e, exc_msg)
+            raise InvalidParameterException(e)
     except Exception as e:
         # Set error message; job_status_id will be set in download_sqs_worker.handle()
-        exc_msg = "An exception was raised while attempting to process the DownloadJob"
-        fail_download(download_job, e, exc_msg)
-        raise Exception(download_job.error_message) from e
+        with SubprocessTrace(
+            name=f"job.{JOB_TYPE}.process_download_job",
+            kind=SpanKind.INTERNAL,
+            service="bulk-download",
+        ) as error_span:
+            error_span.set_attributes(
+                {
+                    "service": "bulk-download",
+                    "span_type": "Internal",
+                    "message": "An exception was raised while attempting to process the DownloadJob",
+                    "error": str(e),
+                }
+            )
+            exc_msg = "An exception was raised while attempting to process the DownloadJob"
```

Comment:
Not worth blocking, but to help consistency I would use this variable in the `error_span` as well instead of copy / paste the same value

Reply:
Reused variable. Done.